spec/models/post_spec.rb

```diff
@@ -35,5 +35,16 @@
         expect(post.up_votes).to eq(1)
       end
     end
+
+    describe "#save_with_initial_vote" do
+      it "saves a post only if the initial vote saves as well" do
+        post = described_class.new(title: "Check One Two",
+                        body:  "Sibilance. Sibilance.",
+                        user:  authenticated_user,
+                        created_at: Time.new)
```

Comment:
@Enriikke for this to work:

```
expect(post.save_with_initial_vote).to eq true
```

we would need to modify the `save_with_initial_vote` method to return a boolean, yes? The way I've currently got it written, it will return a `Vote` so the line would be something like:

```
expect(post.save_with_initial_vote).to be_a Vote
```